### PR TITLE
Fix NPE in NewLineRule

### LINKED ISSUE #789: NPE: Internal error has occurred in newlines
## Describe the bug
On this code:
```kotlin
fun spread(jsObject: Any, handler: (key: String, value: Any) -> Unit) {
}
```
diktat crashes with the following exception:
```
[ERROR] 2021-03-03 15:55:52 Internal error has occurred in newlines. Please make an issue on this bug at https://github.com/cqfn/diKTat/.
java.lang.NullPointerException: it must not be null
	at org.cqfn.diktat.ruleset.rules.chapter3.files.NewlinesRule$handleFirstValue$$inlined$let$lambda$1.invoke(NewlinesRule.kt:360)
	at org.cqfn.diktat.ruleset.rules.chapter3.files.NewlinesRule$handleFirstValue$$inlined$let$lambda$1.invoke(NewlinesRule.kt:100)
```

## Environment information
* diktat version: 0.4.1

### PR DESCRIPTION
**Pull request by Cheshiriks**
### What's done:
* Fixed NPE in NewLineRule for lambda list argument
closes #789

### CODE REVIEW

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/files/NewlinesRule.kt [hunk 1]
```diff
@@ -400,12 +401,14 @@ class NewlinesRule(configRules: List<RulesConfig>) : DiktatRule(
                 warnText
             }
             WRONG_NEWLINES.warnAndFix(configRules, emitWarn, isFixMode, freeText, node.startOffset, node) {
-                node.appendNewlineMergingWhiteSpace(
-                    list.first()
-                        .treePrev
-                        .takeIf { it.elementType == WHITE_SPACE },
-                    list.first()
-                )
+                if (list.first().treePrev != null) {
```
Comment (akuleshov7) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/files/NewlinesRule.kt:404]: @Cheshiriks no tests, no merge of PRs? :)

Comment (akuleshov7) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/files/NewlinesRule.kt:404]: why not avoid null check? Please see https://github.com/cqfn/diKTat/blob/master/info/guide/diktat-coding-convention.md#-433-null-safety

**Review by akuleshov7 — CHANGES_REQUESTED** [review]
please add tests and fix null issues :)

File: diktat-rules/src/test/resources/test/paragraph3/newlines/ListArgumentLambdaTest.kt [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package test.paragraph3.newlines
+
+class Example {
+
+    fun foo() {
+        foldIndexed("#") { index: Int, acc: String, pathPart: String ->
```
Comment (petertrr) [inline diktat-rules/src/test/resources/test/paragraph3/newlines/ListArgumentLambdaTest.kt:6]: I believe this case should be an exception: we should either allow multiple args on the same line, or move the first too. Let's discuss. Also, please add a test with 3 args and no explicit types.

**Review by petertrr — CHANGES_REQUESTED** [review]

**Review by petertrr — APPROVED** [review]
lgtm, just fix code style
